File: PLANTILLAS PYTHON/PlantillaListas.py
class Lista:
    def __init__(self):
        self.items = []

    def agregar(self, elemento):
        self.items.append(elemento)

    def eliminar(self, indice):
        if 0 <= indice < len(self.items):
            eliminado = self.items.pop(indice)
            print(f">> Se eliminó: {eliminado}")

    def buscar(self, valor):
        for i, elemento in enumerate(self.items):
            if elemento == valor:
                return i
        return -1                              # -1 = no encontrado

    def modificar(self, indice, nuevo_valor):
        if 0 <= indice < len(self.items):
            self.items[indice] = nuevo_valor

    def mostrar(self):
        for i, elemento in enumerate(self.items):
            print(f"[{i}] {elemento}")


if __name__ == "__main__":
    mi_lista = Lista()
    mi_lista.agregar("dato 1")
    mi_lista.agregar("dato 2")
    mi_lista.mostrar()
    print("Índice de 'dato 2':", mi_lista.buscar("dato 2"))

File: PLANTILLAS TYPESCRIPT/PlantillaListas.ts
class Lista<T> {
  private items: T[] = [];

  agregar(elemento: T): void {
    this.items.push(elemento);
  }

  eliminar(indice: number): void {
    if (indice >= 0 && indice < this.items.length) {
      const eliminado = this.items.splice(indice, 1)[0];
      console.log(`>> Se eliminó: ${eliminado}`);
    }
  }

  buscar(valor: T): number {
    for (let i = 0; i < this.items.length; i++) {
      if (this.items[i] === valor) {
        return i;
      }
    }
    return -1;                              // -1 = no encontrado
  }

  modificar(indice: number, nuevoValor: T): void {
    if (indice >= 0 && indice < this.items.length) {
      this.items[indice] = nuevoValor;
    }
  }

  mostrar(): void {
    for (let i = 0; i < this.items.length; i++) {
      console.log(`[${i}] ${this.items[i]}`);
    }
  }
}

// ---- USO ----
const miLista = new Lista<string>();
miLista.agregar("dato 1");
miLista.agregar("dato 2");
miLista.mostrar();
console.log("Índice de 'dato 2':", miLista.buscar("dato 2"));